### PR TITLE
Adds transformations notebook

### LINKED ISSUE #514: Add introductory notebook for transformations
### Feature description

This was missed from the initial PR (#357). This notebook should contains the following:

- What are the currently supported transformations
- How to interact with the transformations class (parameters / costs / optimisers)
- When should transformation be used? (large magnitude between inference parameters)
- What care should be taken when using transformations (i.e. [Jensens' inequality](https://en.wikipedia.org/wiki/Jensen%27s_inequality))

### Motivation

_No response_

### Possible implementation

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by BradyPlanden**
# Description

- Adds `apply_transform` functionality to plot2d 
- Adds introductory transformations notebook.

## Issue reference
Fixes #514 

## Review
Before you mark your PR as ready for review, please ensure that you've considered the following:
- Updated the [CHANGELOG.md](https://github.com/pybop-team/PyBOP/blob/develop/CHANGELOG.md) in reverse chronological order (newest at the top) with a concise description of the changes, including the PR number.
- Noted any breaking changes, including details on how it might impact existing functionality.

## Type of change
- [ ] New Feature: A non-breaking change that adds new functionality.
- [ ] Optimization: A code change that improves performance.
- [X] Examples: A change to existing or additional examples.
- [ ] Bug Fix: A non-breaking change that addresses an issue.
- [ ] Documentation: Updates to documentation or new documentation for new features.
- [ ] Refactoring: Non-functional changes that improve the codebase.
- [ ] Style: Non-functional changes related to code style (formatting, naming, etc).
- [ ] Testing: Additional tests to improve coverage or confirm functionality.
- [ ] Other: (Insert description of change)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybop-team/PyBOP/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All unit tests pass: `$ nox -s tests`
- [ ] The documentation builds: `$ nox -s doctest`

You can run integration tests, unit tests, and doctests together at once, using `$ nox -s quick`.

## Further checks:
- [ ] Code is well-commented, especially in complex or unclear areas.
- [ ] Added tests that prove my fix is effective or that my feature works.
- [ ] Checked that coverage remains or improves, and added tests if necessary to maintain or increase coverage.

Thank you for contributing to our project! Your efforts help us to deliver great software.

### CODE REVIEW

File: pybop/plotting/plot2d.py [hunk 1]
```diff
@@ -97,22 +102,28 @@ def plot2d(
     # Populate cost matrix
     for i, xi in enumerate(x):
         for j, yj in enumerate(y):
-            costs[j, i] = cost(np.asarray([xi, yj] + additional_values))
+            costs[j, i] = cost_call(
```
Comment (martinjrobins) [inline pybop/plotting/plot2d.py:105]: Is this right? should you apply the transform in this cost call?

**Review by martinjrobins — CHANGES_REQUESTED** [review]
I got a bit confused as to why you were manually transforming the bounds in the notebook, is this something plot2d should do for you if `apply_transform` is true?

File: pybop/plotting/plot2d.py (hunk 1 above)
Comment (BradyPlanden) [reply pybop/plotting/plot2d.py:105]: This is using the partial function `cost_call()` method which has the transform applied.

**Review by martinjrobins — APPROVED** [review]
looks great, thanks @BradyPlanden

**Pull request merged**